### PR TITLE
Enable install under poetry

### PR DESCRIPTION
**Pull request by simonseed**
Implement numpy hack in setup.py to enable install under Poetry #3363

This is the same as https://github.com/RaRe-Technologies/gensim/pull/3363 but on the 3.8.3 branch

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -108,7 +108,15 @@ def finalize_options(self):
         build_ext.finalize_options(self)
         # Prevent numpy from thinking it is still in its setup process:
         # https://docs.python.org/2/library/__builtin__.html#module-__builtin__
-        __builtins__.__NUMPY_SETUP__ = False
+        try:
+            __builtins__.__NUMPY_SETUP__ = False
+        except:
+            try:
+                # For python 3
```
Comment (piskvorky) [inline setup.py:115]: Gensim does not support python 2 any more.

**Review by piskvorky — CHANGES_REQUESTED** [review]

File: setup.py (hunk 1 above)
Comment (simonseed) [reply setup.py:115]: This is a PR against 3.8.3 which does support Python 2. I am happy to remove Python 2 support from the PR if that is an issue. The problem this solves is, install with Poetry (Python 3) fails for the the same reason as reference in   https://github.com/RaRe-Technologies/gensim/pull/3363

Comment (piskvorky) [reply setup.py:115]: Yes, correct of course.

Comment (simonseed) [reply setup.py:115]: I know it's not supported. But the Poetry install suddenly stopped working - same as in the develop branch. And for those of us that are still using it. This small PR fixes the issue.

Comment (mpenkov) [reply setup.py:115]: I think the best we can do is fix this on develop so future versions won't be affected.

Releasing a bugfix for 3.8.3 requires effort, the volume of which isn't justified, IMHO.

**Comment by mpenkov** [comment]
Closing, will not fix.

**Pull request closed without merging**